lib/index.js

```diff
@@ -72,22 +72,50 @@ var util = require('./common/util.js');
  * your provided configuration will remain isolated to the returned gcloud
  * module.
  *
+ * To authenticate API requests,
+ * [google-auth-library](https://github.com/google/google-auth-library-nodejs)
+ * is used to detect the environment your project is running in. Use the
+ * following guide to determine the appropriate configuration.
+ *
+ * <h4>Google Compute Engine</h4>
+ *
+ *   - No configuration necessary!
+ *
+ * <h4>Google App Engine Production</h4>
+ *
+ *   - No configuration necessary!
+ *
+ * <h4>Other</h4>
+
+ *   - Provide `config.projectId`
+ *   - Use one of the following:
+ *     - `config.credentials` object containing `client_email` and `private_key`
+ *       properties.
+ *     - `config.keyFilename` path to a key file.
```

Comment:
Say here what extensions are supported. 

`config.keyFilename` path to a key file. Supported types: .json, .pem* and .p12*